### PR TITLE
Fix buggy pokemon anims with Illusion 

### LINKED ISSUE #2624: Pokémon anims are buggy with illusion
Enemy Party:
Zoroark
Mon 2
Mon 2

When enemy sends out Zoroark, and it's disguised as, an example being a Politoed, the Politoed will play the Zoroark sprite anim (the one that flickers and shows/hide briefly).
So 2 problems:
- Loading wrong anim, so it looks weird/broken
- Player knows that its a Zoroark because the different anim being played.

### PR DESCRIPTION
**Pull request by DizzyEggg**
Fixes #2624

There is an other Illusion issue opened atm, but it will be done in a separate PR(if and when no idea :D )

### CODE REVIEW

**Comment by Jaizu** [comment]
If this is not a draft anymore restore Lotad abilities :D

File: src/data/pokemon/species_info.h [hunk 1]
```diff
@@ -7063,7 +7063,7 @@ const struct SpeciesInfo gSpeciesInfo[] =
         .friendship = 70,
         .growthRate = GROWTH_MEDIUM_SLOW,
         .eggGroups = { EGG_GROUP_WATER_1, EGG_GROUP_GRASS},
-        .abilities = {ABILITY_SWIFT_SWIM, ABILITY_RAIN_DISH, ABILITY_OWN_TEMPO},
+        .abilities = {ABILITY_ILLUSION, ABILITY_ILLUSION},
```
Comment (Jaizu) [inline src/data/pokemon/species_info.h:7066]: You should avoid submiting changes that you made for testing.
Also, revert since it's not a draft anymore o/

**Review by Jaizu — CHANGES_REQUESTED** [review]

File: src/battle_util.c [hunk 2]
```diff
@@ -10266,6 +10266,14 @@ void ClearIllusionMon(u32 battlerId)
     memset(&gBattleStruct->illusion[battlerId], 0, sizeof(gBattleStruct->illusion[battlerId]));
 }
 
+u32 GetIllusionMonSpecies(u32 battlerId)
+{
+    struct Pokemon *illusionMon = GetIllusionMonPtr(battlerId);
+    if (illusionMon != NULL)
+        return GetMonData(illusionMon, MON_DATA_SPECIES);
+    return 0;
```
Comment (sphericalice) [inline src/battle_util.c:10274]:
```suggestion
    return SPECIES_NONE;
```

**Comment by DizzyEggg** [comment]
Ready for review

**Review by Jaizu — APPROVED** [review]

**Pull request merged**